tool/src/main/java/org/wildfly/security/tool/FileSystemEncryptRealmCommand.java

```diff
@@ -0,0 +1,737 @@
+                            descriptor.setEncoded(Boolean.parseBoolean(arg));
+                            break;
+                        case LEVELS_PARAM:
+                            descriptor.setLevels(Integer.parseInt(arg));
+                            break;
+                    }
+                }
+            });
+        } catch (IOException e) {
+            errorHandler(e);
+        }
+        int currentCount = count.intValue();
+        findMissingRequiredValuesAndSetValues(currentCount, descriptor);
+        copyAddResetDescriptor(descriptor);
+        if (summaryMode) {
+            printDescriptorBlocks(currentCount);
+        }
+        count.getAndIncrement();
+    }
+
+    /**
+     * Copies a Descriptor into a temporary one for it can be cloned into the list of descriptors
+     *
+     * @param original The original descriptor that is continually modified
+     */
+    private void copyAddResetDescriptor(Descriptor original) {
+        Descriptor temp = new Descriptor(original);
+        descriptors.add(temp);
+        original.reset();
+    }
+
+    /**
+     * Determines if the current descriptor block is missing any required values
+     * and then if it is, prints out a warning message and sets that required value
+     * and the optional values to null
+     *
+     * @param count The index of the current descriptor block
+     */
+    private void findMissingRequiredValuesAndSetValues(int count, Descriptor descriptor) {
+        boolean missingRequiredValue = false;
+        if (descriptor.getInputRealmLocation() == null) {
+            warningHandler(ElytronToolMessages.msg.skippingDescriptorBlockInputLocation(count));
+            missingRequiredValue = true;
+        }
+        if (descriptor.getOutputRealmLocation() == null) {
+            warningHandler(ElytronToolMessages.msg.skippingDescriptorBlockOutputLocation(count));
+            missingRequiredValue = true;
+        }
+        if (descriptor.getFileSystemRealmName() == null) {
+            warningHandler(ElytronToolMessages.msg.skippingDescriptorBlockFilesystemRealmName(count));
+            missingRequiredValue = true;
+        }
+        if(descriptor.getHashCharset() == null) {
+            descriptor.setHashCharset(StandardCharsets.UTF_8);
+        }
+        if(descriptor.getHashEncoding() == null) {
+            descriptor.setHashEncoding(Encoding.BASE64);
+        }
+        if(descriptor.getEncoded() == null) {
+            descriptor.setEncoded(true);
+        }
+        if(descriptor.getLevels() == 0) {
+            descriptor.setLevels(2);
+        }
+        if (missingRequiredValue) {
+            descriptor.reset();
+        }
+    }
+
+    /**
+     * Handles creating the Elytron filesystem-realm from the descriptor array
+     *
+     * @throws Exception Exception to be handled by Elytron Tool
+     */
+    private void createFileSystemRealm() throws Exception {
+        Security.addProvider(new WildFlyElytronProvider());
```

Comment:
Instead of using this provider, it should be possible to use the specific ones needed.

Reply:
Removed